Poly transcends mortality, asks for crackers.

Incoming5643 opened the pull request:
Poly now tracks death streaks just as he currently does with survival streaks.

Poly's survival streak is now advertised in his description when relevant. The death streak is only advertised when it's a record.

Poly's color varies slightly when he's holding either one of his records.

Killing poly while he's holding either of his records triggers a horrifying easter egg (this still breaks survival streaks and adds to death streaks).

Incoming5643 committed:
Poly now tracks death streaks just as he currently does with survival streaks.

Poly's survival streak is now advertised in his description when relevant. The death streak is only advertised when it's a record.

Killing poly while he's holding either of his records triggers a horrifying easter egg (this still breaks survival streaks and adds to death streaks).

Incoming5643 commented:
:cake: Probably the last bird pull :cake:

**Probably**

KazeEspada commented on code/modules/mob/living/simple_animal/parrot.dm:
```diff
@@ -908,4 +929,18 @@
+	color = "#FFFFFF77"
+	speak_chance = 20
+	status_flags = GODMODE
+	incorporeal_move = 1
```
Does his targetting even allow this?

Incoming5643 replied:
A* doesn't account for incorpreal move but poly can still get in situations where he could use it because he looks for paths with A*, but then moves straight to whatever he's targeting.

KazeEspada commented on code/modules/mob/living/simple_animal/parrot.dm:
```diff
@@ -908,4 +929,18 @@
+	desc = "Doomed to squawk the earth."
+	color = "#FFFFFF77"
+	speak_chance = 20
+	status_flags = GODMODE
```
Let's find out how many bugs are with GODMODE.

Incoming5643 replied:
It might not catch gibbing/dusting or the wand of death but outside of that it's actually surprisingly stable.

By design the ghost of poly will usually be with the one person on the station who could easily dispose of the ghost of poly even with GODMODE on.

PKPenguin321 commented:
Wake me up

optimumtact commented:
The kind of neat little well coded mini features that make the game fun

GunHog commented:
I really wonder what Tokiko thinks of all this. Poly has become quite the interesting little pet!

neersighted commented on code/modules/mob/living/simple_animal/parrot.dm:
```diff
@@ -874,24 +875,43 @@
+		color = "#EEEE22"
+	else if(rounds_survived == longest_deathstreak)
+		speak += pick("What are you waiting for!", "Violence breeds violence!", "Blood! Blood!", "Strike me down if you dare!")
+		desc += " The squawks of [rounds_survived * -1] dead parrots ring out in your ears..."
```
Why is this a negative number?

RemieRichards replied:
it reuses the ```rounds_survived``` var (if you've died twice in a row, you've survived -2 rounds) so it's converted to a positive number here, so that it's not "The squawks of -2 dead parrots ring out in your ears..."

duncathan replied:
Why not just `[-rounds_survived]`?

Incoming5643 replied:
I wasn't 100% sure that would work, even though it should work.

Because you know, byond.

neersighted commented on code/modules/mob/living/simple_animal/parrot.dm:
```diff
@@ -908,4 +929,18 @@
+	speak_chance = 20
+	status_flags = GODMODE
+	incorporeal_move = 1
+	butcher_results = list(/obj/item/weapon/ectoplasm/ = 1)
```
Trailing slash

tkdrg replied:
This could be fixed I guess.

AlexanderUlanH commented:
I, for one, will enjoy the ensuing metawars over Poly's survival.

ChangelingRain commented:
Squawk.

tkdrg commented on code/modules/mob/living/simple_animal/parrot.dm:
```diff
@@ -874,24 +875,43 @@
+			if(mind)
+				mind.transfer_to(G)
+			else
+				G.key = key
```
We really need a proc to encapsulate these four lines, it's used everywhere and very bug prone. You don't have to do it here though, I do realize that this is asking too much.

kevinz000 commented:
so why again can poly recite shit from innumerable rounds ago :/

Incoming5643 commented:
Because he's transcended his round based existence but still trapped by it, leaving him in a horrifying state of hopelessness where nothing really matters.

Also it's funny and can lead to fun situations for very little cost.

GunHog commented:
>Because he's transcended his round based existence but still trapped by it, leaving him in a horrifying state of hopelessness where nothing really matters.

So...Poly is Sans?

Incoming5643 commented:
Unfortunately, yes.

Incoming5643 committed:
Take cares of minor suggested code tweaks

Incoming5643 commented:
Provided this compiles we should be good to go.

Dorsisdwarf commented:
It's a bit late, but can I ask for a var on the probbability of Poly forgetting lines once said?
Setting speakchance to 50 is great fun, but they repeat themselves 10-12 times before deleting words

Incoming5643 commented:
Here's the thing about how poly shuffles his lines: He has a chance of swapping one line from the 500 line buffer to the 4~6 active recall list every life tick. This happens regardless of if he's actually saying something or not. The repetition only really sets in if poly's speak chance raises significantly above his effective shuffle rate.

I'll add a var for this, but I'm going to tie it to his speak chance too so he shuffles more actively when he talks more actively.

Incoming5643 committed:
adds speech_shuffle_rate as a var

speech_shuffle_rate will also raise when poly is fed a cracker to help cut down on line repetition

tkdrg commented:
@Incoming5643 parrot conflicts

Incoming5643 commented:
It's because you **just** merged https://github.com/tgstation/-tg-station/pull/15490

Gah give me a moment

Incoming5643 committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into I_dont_know_why_Im_doing_this_but_its_not_your_job_to_stop_me

# Conflicts:
#	code/modules/mob/living/simple_animal/parrot.dm

Incoming5643 commented:
@tkdrg @Dorsisdwarf should be fine once travis clears
